### PR TITLE
Fix guice dependency problem

### PR DESCRIPTION
**Pull request by xsalefter**

### CODE REVIEW

File: profiles/killbill/pom.xml [hunk 1]
```diff
@@ -61,15 +61,25 @@
...
             <groupId>com.google.code.findbugs</groupId>
             <artifactId>jsr305</artifactId>
-            <scope>provided</scope>
+            <scope>compile</scope>
         </dependency>
         <dependency>
             <groupId>com.google.inject</groupId>
             <artifactId>guice</artifactId>
-            <scope>provided</scope>
+            <scope>compile</scope>
```
Comment (sbrossie) [inline profiles/killbill/pom.xml:82]: That seems reasonable - who was 'providing' such package before this?

**Review by sbrossie — APPROVED** [review]
I tried launching KB with the change using the `jetty:run` and it seems to be working fine on my end.

File: profiles/killbill/pom.xml (hunk 1 above)
Comment (pierre) [reply profiles/killbill/pom.xml:82]: FYI `compile` is the default scope (and it isn't overridden in the base pom.xml: https://github.com/killbill/killbill-oss-parent/blob/17b5048cafb6fe6c80a58ce12c6163844dea214c/pom.xml#L361-L365), so you shouldn't even need a scope at all.

File: pom.xml [hunk 2]
```diff
@@ -59,11 +59,12 @@
         <url>http://github.com/killbill/killbill</url>
     </issueManagement>
     <properties>
+        <check.skip-dependency-versions>true</check.skip-dependency-versions>
```
Comment (pierre) [inline pom.xml:62]: Let's not forget to fix this eventually...

File: pom.xml [hunk 3]
```diff
@@ -106,43 +107,6 @@
                     </exclusion>
                 </exclusions>
             </dependency>
-            <!-- Temporary until 0.24.x is released -->
```
Comment (pierre) [inline pom.xml:109]: Wondering if I had added this specifically for that `check.skip-dependency-versions` 🤔 

I think even the `two-digits-backward-compatible` strategy isn't good enough for our `SNAPSHOT` scheme: https://github.com/killbill/killbill-oss-parent/blob/17b5048cafb6fe6c80a58ce12c6163844dea214c/pom.xml#L1712

The real fix would be:

1. Switch to https://github.com/basepom/dependency-versions-check-maven-plugin from the old Ning plugin (see https://github.com/killbill/killbill-oss-parent/pull/445 and https://github.com/killbill/killbill-oss-parent/pull/431#issuecomment-1112488332 for why I couldn't do it)
2. Submit a PR upstream to @hgschmie to support our naming scheme (`ignore-SNAPSHOT-with-sha` strategy)

Maybe take a look at this once the build is stable again?

File: profiles/killbill/src/main/java/org/killbill/billing/server/modules/KillbillServerModule.java [hunk 4]
```diff
@@ -106,6 +116,15 @@ protected void configure() {
         configurePushNotification();
 
         bind(new TypeLiteral<OSGIServiceRegistration<Healthcheck>>() {}).to(DefaultHealthcheckPluginRegistry.class).asEagerSingleton();
+
+        // Needed because changes in KillbillHealthcheck from Guice @Inject(optional=true) to javax @Inject + @Nullable
```
Comment (pierre) [inline profiles/killbill/src/main/java/org/killbill/billing/server/modules/KillbillServerModule.java:120]: Subtle! 😅 

We should do some additional manual testing to verify all of this plumbing still works as expected. For instance, service discovery can be tested using the Eureka bundle (https://github.com/killbill/killbill-platform/tree/work-for-release-0.23.x/osgi-bundles/bundles/eureka). @vnandwana can help if needed (he had to do similar testing when doing the metrics work).

**Review by pierre — APPROVED** [review]

File: profiles/killbill/pom.xml (hunk 1 above)
Comment (xsalefter) [reply profiles/killbill/pom.xml:82]: > That seems reasonable - who was 'providing' such package before this?

[I did](https://github.com/killbill/killbill/commit/429ac9b1166f724cbd18a953279e6400be3fa91d#diff-030ec92e9612718b2d2bc8aa5b7a65ebd296c063e121df61c273b3b00530b926) :man_facepalming: . I think in attempt to fix something else.

File: profiles/killbill/src/main/java/org/killbill/billing/server/modules/KillbillServerModule.java (hunk 4 above)
Comment (xsalefter) [reply profiles/killbill/src/main/java/org/killbill/billing/server/modules/KillbillServerModule.java:120]: Noted.

File: pom.xml (hunk 3 above)
Comment (xsalefter) [reply pom.xml:109]: Tracked [here](https://github.com/killbill/killbill/issues/1763). Meanwhile, I'll mark this as resolved.

File: profiles/killbill/pom.xml (hunk 1 above)
Comment (xsalefter) [reply profiles/killbill/pom.xml]: > so you shouldn't even need a scope at all.

Fixed by this [commit](https://github.com/killbill/killbill/pull/1760/commits/aaedb5e4f3755ad87e74df93f946518e186d85e0).

File: profiles/killbill/src/main/java/org/killbill/billing/server/modules/KillbillServerModule.java (hunk 4 above)
Comment (xsalefter) [reply profiles/killbill/src/main/java/org/killbill/billing/server/modules/KillbillServerModule.java:120]: @pierre tracked [here](https://github.com/killbill/killbill/issues/1764), although I'm not sure what do you mean by ".... service discovery can be tested using the Eureka bundle".

**Comment by sbrossie** [comment]
Merging as this solves the issue with the build 🎉

**Pull request merged**